nwbinspector/checks/time_series.py

```diff
@@ -92,3 +92,38 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=TimeSeries)
+def check_rows_not_nan(time_series: TimeSeries, nelems=200):
+    """Check that each row of a TimeSeries has at least one non-NaN piece of data."""
+    n_dims = len(time_series.data.shape)
+    if n_dims > 2:
+        yield
+
+    subframe_selection = np.unique(
+        np.round(np.linspace(start=0, stop=time_series.data.shape[0] - 1, num=nelems)).astype(int)
+    )
+    if n_dims == 1:
+        if not all(np.isnan(time_series.data[:nelems])):
+            yield
+
+        if all(np.isnan(time_series.data[subframe_selection])):
+            yield InspectorMessage(
+                message=(
+                    "This TimeSeries appears to contain NaN data at each frame. "
+                    "Consider removing this object from the file."
+                )
+            )
+    elif n_dims == 2:
+        for col in range(time_series.data.shape[1]):
+            if not all(np.isnan(time_series.data[:nelems, col]).flatten()):
+                continue
+
+            if all(np.isnan(time_series.data[subframe_selection, col]).flatten()):
```

Comment:
If there a way of modifying this code to minimize the number of h5py dataset reads?

Reply:
Yeah, we could take the approach of pulling out and checking data in packets of 'chunks' (would be much more efficient for streaming).

In the worst case, as it currently stands, the slicing might span `nelems` total number of chunks for each column, which *might* be cached for a bit on the HDF5 end but unlikely that's much benefit at normal data scales. Either way is definitely a bit inefficient for that.

One idea could be to check if the `time_series` being read is chunked, and if it is, alter our interpretation of `nelems` from spanning `nelems` individual data values to spanning some smaller number of chunks throughout the dataset, but utilize all of the data retrieved from those chunks. 

This is also all separate from the first early exit condition of this entire check, which only examines the first `nelems` consecutive data values, which are fairly unlikely to span multiple chunks unless the `chunk_shape` is something extreme like by-frame for all channels.

I can think more about the total data usage that this function might require and report some approximate numbers. But yeah, we might need to be a bit more restrictive than usual with the `nelems` subsetting, the interaction with chunking could give some poor performance in practice.